### PR TITLE
Course model: Part 1

### PR DESCRIPTION
**Pull request by Bjvanminnen**
New attempt at https://github.com/code-dot-org/code-dot-org/pull/14805/files based on feedback.

This comes in three parts (this is Part 1). Part 1 needs to be in a separate DTP, since changes to the model in future parts cause the migration not work. Parts 2 and 3 could theoretically be combined.

**(1) Creates the new table, and populates it with values from Plc::Course table. The model is not used outside of the migration at all.**
(2) Add our association between models, delegate name from Plc::Course to Course, and update code. At this point the name column still exists on Plc::Course table, but should be unused.
https://github.com/code-dot-org/code-dot-org/pull/14852
(3) Our only destructive migration, this just removes the name column.
https://github.com/code-dot-org/code-dot-org/pull/14853

I've validated that at each step along the way dashboard tests pass.

Future work (after parts 1 - 3):
Update routes.rb to have /course/:courseId go to a courses controller. This controller will then route plc courses to the appropriate plc controller
Another migration that adds a CourseScript model (representing the scripts that are in a non-plc course)

### CODE REVIEW

File: dashboard/test/models/course_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class CourseTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (aoby) [inline dashboard/test/models/course_test.rb:6]: How will we know if the truth is still true when this test is commented-out ;)

**Comment by aoby** [comment]
You'll (clearly) need to resolve `schema.rb` conflicts

File: dashboard/test/models/course_test.rb (hunk 1 above)
Comment (Bjvanminnen) [reply dashboard/test/models/course_test.rb:6]: I've found commenting out tests to be one of the most effective ways of getting them to pass.

Comment (aoby) [reply dashboard/test/models/course_test.rb:6]: More or less effective than `assert true`?

File: dashboard/app/models/course.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# == Schema Information
+#
+# Table name: courses
+#
+#  id            :integer          not null, primary key
+#  name          :string(255)
+#  properties    :text(65535)
+#  plc_course_id :integer
```
Comment (mehalshah) [inline dashboard/app/models/course.rb]: Thinking a little more - why not remove plc_course_id and add course_id to plc_courses?

Comment (mehalshah) [reply dashboard/app/models/course.rb]: It's more intuitive with the hierarchy in my head

Comment (Bjvanminnen) [reply dashboard/app/models/course.rb]: I think I agree, and did this having misunderstood a comment on my RFC PR.

File: dashboard/test/models/course_test.rb (hunk 1 above)
Comment (ashercodeorg) [reply dashboard/test/models/course_test.rb:6]: Definitely. In practice, I've had `assert true` tests fail.

Rails Interview Question: For what possible reasons might
```
test 'the truth' do
  assert true
end
```
fail?

Comment (aoby) [reply dashboard/test/models/course_test.rb:6]: I'd say if you redefine `true`, but that doesn't seem to be allowed. Even if I define a method called `true`, it's not called. You could blow up by having another test with the same name or something, but I'm not sure how `assert true` itself could fail. Please do tell :)

Comment (Bjvanminnen) [reply dashboard/test/models/course_test.rb:6]: Could you redefine assert?

Comment (Bjvanminnen) [reply dashboard/test/models/course_test.rb:6]: Or test for that matter?

Comment (aoby) [reply dashboard/test/models/course_test.rb:6]: You could do something like this:

```
irb(main):019:0> class TrueClass
irb(main):020:1>   def ==(value)
irb(main):021:2>     false
irb(main):022:2>   end
irb(main):023:1> end
=> :==
irb(main):024:0> true == true
=> false
```

**Comment by Bjvanminnen** [comment]
Updated to make it so that `Plc::Course` has the foreign_key instead of `Course`. At this point, Parts 2 and 3 are out of date, but trying to maintain all three parts at once has shown itself to be difficult. I'd like to get Part 1 merged, and then focus on Part 2.

**Pull request merged**